### DISCUSSION TITLE
Paste icon types

### DISCUSSION THREAD

**Discussion by crubio-imp** [twilio-labs/paste#593] [General] [answered]
I am rendering an icon based on the status of a sim and know (for now) it will be one of three types
Is there a helper or generic type i can use to cover any of these individual icon types?

Code for reference:

`type SuperSimIconType = React.FC<SuccessIconProps | ErrorIconProps | InformationIconProps>;
export interface SuperSimIconParams {
  tagName: SuperSimIconType;
  title: string;
  decorative: boolean;
  iconColor: TextColor;
  size: IconSize;
}`

**Comment by serifluous** [comment]
Thanks, @crubio-imp! We're looking into this and will get back to you within a day.

**Comment by TheSisb** [comment, marked as answer]
Hi @crubio-imp 

As of now there isn't a generic type you could use because we're still holding on to the possibility of having certain icons with different interfaces. Feel free to create your own generic type in your code and use that for now.
i.e.: `interface IconProps extends SuccessIconProps {}` or similar
